Source/WebCore/rendering/HitTestResult.cpp

```diff
@@ -542,6 +542,31 @@ void HitTestResult::enterFullscreenForVideo() const
 #endif
 }
 
+bool HitTestResult::mediaIsInVideoViewer() const
+{
+#if PLATFORM(MAC) && ENABLE(VIDEO) && ENABLE(VIDEO_PRESENTATION_MODE)
+    HTMLMediaElement* mediaElt(mediaElement());
+    return is<HTMLVideoElement>(mediaElt) && mediaElt->fullscreenMode() == HTMLMediaElementEnums::VideoFullscreenModeInWindow;
+#else
+    return false;
+#endif
+}
+
+void HitTestResult::toggleVideoViewer() const
+{
+#if PLATFORM(MAC) && ENABLE(VIDEO) && ENABLE(VIDEO_PRESENTATION_MODE)
+    RefPtr mediaElement(this->mediaElement());
+    RefPtr videoElement = dynamicDowncast<HTMLVideoElement>(mediaElement);
```

Comment:
Nits: Early returns are always ok, so this could bail after the first `RefPtr`. Also, this could be written either as a simple assignment, or with the modern, curly-brace initializer:

```
RefPtr mediaElement { this->mediaElement() };
if (!mediaElement)
    return;

RefPtr videoElement = dynamicDowncast<HTMLVideoElement>(mediaElement);
```

But this _would_ null-check `mediaElement` twice, so you could add back in the dereference to `mediaElement` inside `dynamicDowncast()`. Just a suggestion though.